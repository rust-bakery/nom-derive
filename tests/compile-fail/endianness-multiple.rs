extern crate nom;
extern crate nom_derive;
use nom_derive::*;

#[derive(Nom)]
#[nom(BigEndian, LittleEndian)] // ERROR: Struct cannot be both big and little endian
pub struct BothEndian1 {
    #[nom(Parse="u32")]
    a: u32,
}

#[derive(NomLE)] // ERROR: Object cannot be both big and little endian
#[nom(BigEndian)]
pub struct BothEndian2 {
    a: u32,
}

fn main() {}
