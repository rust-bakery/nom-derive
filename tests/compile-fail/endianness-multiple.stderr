error: Struct cannot be both big and little endian
 --> $DIR/endianness-multiple.rs:6:7
  |
6 | #[nom(BigEndian, LittleEndian)] // ERROR: Struct cannot be both big and little endian
  |       ^^^^^^^^^

error: Object cannot be both big and little endian
  --> $DIR/endianness-multiple.rs:12:10
   |
12 | #[derive(NomLE)] // ERROR: Object cannot be both big and little endian
   |          ^^^^^
   |
   = note: this error originates in the derive macro `NomLE` (in Nightly builds, run with -Z macro-backtrace for more info)
